src/main/java/emu/grasscutter/game/managers/blossom/BlossomManager.java

```diff
@@ -0,0 +1,237 @@
+package emu.grasscutter.game.managers.blossom;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import emu.grasscutter.Grasscutter;
+import emu.grasscutter.data.GameData;
+import emu.grasscutter.data.GameDepot;
+import emu.grasscutter.data.common.ItemParamData;
+import emu.grasscutter.data.excels.RewardPreviewData;
+import emu.grasscutter.game.entity.EntityGadget;
+import emu.grasscutter.game.entity.GameEntity;
+import emu.grasscutter.game.entity.gadget.GadgetWorktop;
+import emu.grasscutter.game.inventory.GameItem;
+import emu.grasscutter.game.player.Player;
+import emu.grasscutter.game.world.Scene;
+import emu.grasscutter.game.world.SpawnDataEntry;
+import emu.grasscutter.game.world.SpawnDataEntry.SpawnGroupEntry;
+import emu.grasscutter.net.proto.BlossomBriefInfoOuterClass;
+import emu.grasscutter.net.proto.VisionTypeOuterClass;
+import emu.grasscutter.server.packet.send.PacketBlossomBriefInfoNotify;
+import emu.grasscutter.utils.Utils;
+import it.unimi.dsi.fastutil.ints.Int2ObjectLinkedOpenHashMap;
+import it.unimi.dsi.fastutil.ints.Int2ObjectMap;
+import it.unimi.dsi.fastutil.ints.IntArrayList;
+import it.unimi.dsi.fastutil.ints.IntList;
+
+public class BlossomManager {
+    public BlossomManager(Scene scene) {
+        this.scene = scene;
+    }
+
+    private final Scene scene;
+    private final List<BlossomActivity> blossomActivities = new ArrayList<>();
+    private final List<BlossomActivity> activeChests = new ArrayList<>();
+    private final List<EntityGadget> createdEntity = new ArrayList<>();
+
+    private final List<SpawnDataEntry> blossomConsumed = new ArrayList<>();
+
+    public void onTick(){
+        synchronized (blossomActivities){
+            var it = blossomActivities.iterator();
+            while(it.hasNext()){
+                var active = it.next();
```

Comment:
Not a fan of these iterator patterns, not sure if there's a cleaner way of doing iterate and delete though.